### PR TITLE
feat: add emr on eks managed endpoint support

### PR DESCRIPTION
**Pull request by lmouhib**
**Issue #, if available:**

#360

## Description of changes:

Add support for [interactive endpoint](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/how-it-works.html)

**Checklist**

 

* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/) (`fix: `, `feat: `, `docs: `, ...)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: framework/src/processing/lib/emr-releases.ts [hunk 1]
```diff
@@ -44,6 +66,7 @@ export enum SparkImage {
 }
 
 export const EMR_DEFAULT_VERSION: EmrRuntimeVersion = EmrRuntimeVersion.V6_15;
+export const EMR_CONTAINETS_DEFAULT_VERSION: EmrContainersRuntimeVersion = EmrContainersRuntimeVersion.V6_15;
```
Comment (vgkowski) [inline framework/src/processing/lib/emr-releases.ts:69]: typo

File: framework/src/processing/lib/spark-job/spark-job-emr-container.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ import { EMR_DEFAULT_VERSION } from '../emr-releases';
  * import { JsonPath } from 'aws-cdk-lib/aws-stepfunctions';
  *
  * const job = new dsf.processing.SparkEmrContainerJob(this, 'SparkJob', {
+ * const job = new dsf.processing.SparkEmrContainerJob(this, 'SparkJob', {
```
Comment (vgkowski) [inline framework/src/processing/lib/spark-job/spark-job-emr-container.ts:25]: duplicated line

File: framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts [hunk 3]
```diff
@@ -86,7 +88,8 @@ export class SparkEmrContainersRuntime extends TrackedConstruct {
   /**
    *  The default EMR on EKS version
    */
-  public static readonly DEFAULT_EMR_EKS_VERSION = EMR_DEFAULT_VERSION;
+  public static readonly DEFAULT_EMR_EKS_VERSION = EMR_CONTAINETS_DEFAULT_VERSION;
```
Comment (vgkowski) [inline framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts:91]: typo

File: framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-interactive-endpoint-props.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { EmrContainersRuntimeVersion } from '../../emr-releases';
+
+/**
+* The properties for the EMR Managed Endpoint to create.
+*/
+export interface SparkEmrContainersRuntimeInteractiveSessionProps {
+  /**
+     * The name of the EMR managed endpoint
```
Comment (vgkowski) [inline framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-interactive-endpoint-props.ts:12]: bad indentation

File: framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts [hunk 5]
```diff
@@ -620,5 +678,41 @@ export class SparkEmrContainersRuntime extends TrackedConstruct {
...
+   * Creates a new Amazon EMR managed endpoint to be used with Amazon EMR Virtual Cluster .
+   * CfnOutput can be customized.
+   * @param {Construct} scope the scope of the stack where managed endpoint is deployed
+   * @param {string} id the CDK id for endpoint
+   * @param {SparkEmrContainersRuntimeInteractiveSessionProps} interactiveSessionOptions the EmrManagedEndpointOptions to configure the Amazon EMR managed endpoint
+   */
+
+  public addInteractiveEndpoint(scope: Construct, id: string, interactiveSessionOptions: SparkEmrContainersRuntimeInteractiveSessionProps) {
+
+    if (interactiveSessionOptions.managedEndpointName.length > 64) {
```
Comment (vgkowski) [inline framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts:692]: Should we add a unique hash at the end of the name? Does it need to be unique?

File: framework/test/unit/processing/spark-runtime-containers.test.ts [hunk 6]
```diff
@@ -940,3 +942,176 @@ describe('With provided EKS cluster, the construct ', () => {
...
+              },
+            ),
+          ]),
+        }),
+        Roles: ['StartJobRole'],
+      }));
+  });
+
+
+});
```
Comment (vgkowski) [inline framework/test/unit/processing/spark-runtime-containers.test.ts:1117]: can you add tests for the RemovalPolicy?  Something similar to this:

```
  test(' create resources with Delete policy', () => {
    template.allResources('AWS::Logs::LogGroup', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });

    template.allResources('Custom::RedshiftDataSql', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });

    template.allResources('Custom::LambdaEniCleanup', {
      UpdateReplacePolicy: 'Delete',
      DeletionPolicy: 'Delete',
    });
  });
```

And same thing for the revert. This is to ensure all resources are deleted properly.

**Review by vgkowski — CHANGES_REQUESTED** [review]

File: framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts (hunk 5 above)
Comment (lmouhib) [reply framework/src/processing/lib/spark-runtime/emr-containers/spark-emr-containers-runtime.ts:692]: no, there is no need for uniqueness, the uniqueness is handled by the physical resource id, managed_endpoint_id.

**Review by vgkowski — APPROVED** [review]

**Pull request merged**